examples/nextjs-spreadsheet-advanced/src/components/Cell.tsx

```diff
@@ -0,0 +1,180 @@
+import cx from "classnames";
+import {
+  type CSSProperties,
+  type ChangeEvent,
+  type ComponentProps,
+  type KeyboardEvent as ReactKeyboardEvent,
+  useCallback,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import { useSelf } from "../liveblocks.config";
+import { EXPRESSION_ERROR } from "../spreadsheet/interpreter/utils";
+import type { UserInfo } from "../types";
+import { appendUnit } from "../utils/appendUnit";
+import { useAutoFocus } from "../utils/useAutoFocus";
+import styles from "./Cell.module.css";
+
+export interface Props extends Omit<ComponentProps<"td">, "onSelect"> {
+  expression: string;
+  getExpression: () => string;
+  height: number;
+  isSelected?: boolean;
+  isEditing?: boolean;
+  onStartEditing: () => void;
+  onEndEditing: () => void;
+  onDelete: () => void;
+  onSelect: () => void;
+  onCommit: (value: string, direction?: "down") => void;
+  other?: UserInfo;
+  width: number;
+}
+
+export interface EditingCellProps extends ComponentProps<"input"> {
+  expression: string;
+  onEndEditing: () => void;
+  onCommit: (value: string, direction?: "down") => void;
+}
+
+export function formatValue(value: string) {
+  return value.replace(" ", "").toUpperCase();
+}
+
+function EditingCell({
+  expression,
+  onCommit,
+  onEndEditing,
+  className,
+  ...props
+}: EditingCellProps) {
+  const input = useRef<HTMLInputElement>(null);
+  const [draft, setDraft] = useState<string>(expression);
+  const value = useMemo(
```

Comment:
@marcbouchenoire you should be able to use `spreadsheet/interpreter/tokenizer` to get an array of tokens to be able to colorize the expression and replace the input with a span with `contentEditable={true}` but it might not be easy. Let me know if you want to pair a bit on this.